Source/FakeItEasy.Specs/DisposableSpecs.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace FakeItEasy.Specs
+{
+    using System;
+    using FluentAssertions;
+    using Machine.Specifications;
+
+    public class when_faking_a_disposable_class
+        : EventRaisingSpecs
+    {
+        static IDisposable fake;
+        private static Exception exception;
+
+        Establish context = () =>
+        {
+            AppDomain.CurrentDomain.UnhandledException += UnhandledExceptionHandler;
+
+            fake = A.Fake<SomeDisposable>();
+        };
+
+        Because of = () =>
+        {
+            fake = null;
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+        };

```

Comment:
But at least, in our current environments, it has proved that the fix works, and that's the most important thing.